### PR TITLE
Array.prototype.includes

### PR DESCRIPTION
**Pull request by NeilFraser**

### CODE REVIEW

File: demo/core_30_$.servers.http.js [hunk 1]
```diff
@@ -99,7 +99,7 @@ $.servers.http.IncomingMessage.prototype.parse = function(line) {
       }
       value = existing;
     } else if (name in this.headers) {
-      if ($.servers.http.IncomingMessage.discardDuplicates.indexOf(name) === -1) {
```
Comment (cpcallen) [inline demo/core_30_$.servers.http.js]: Looks like you're fixing a bug here?

File: demo/core_30_$.servers.http.js [hunk 2]
```diff
@@ -198,7 +198,7 @@ $.servers.http.ServerResponse.prototype.setHeader = function(name, value) {
   } else {
     var existing = this.headers_[name];
     if (name in this.headers_) {
-      if ($.servers.http.ServerResponse.discardDuplicates.indexOf(name) === -1) {
```
Comment (cpcallen) [inline demo/core_30_$.servers.http.js]: Ditto.

File: server/interpreter.js [hunk 3]
```diff
@@ -1106,6 +1106,33 @@ Interpreter.prototype.initArray_ = function() {
...
+      if (len === 0) return false;
+      var n = (fromIndex === undefined ? 0 : Interpreter.toInteger(fromIndex));
+      if (n >= len) return false;
+      var k = (n >= 0) ? n : Math.max(len - Math.abs(n), 0);
+      for (; k < len; k++) {
+        if (obj.has(String(k), perms)) {
+          var v = obj.get(String(k), perms);
+          if (v === searchElement ||
+              (typeof v === 'number' && typeof searchElement === 'number' &&
+               isNaN(v) && isNaN(searchElement))) {
```
Comment (cpcallen) [inline server/interpreter.js]: Use `Number.isNaN` and you can omit the separate typeof tests.

Comment (NeilFraser) [reply server/interpreter.js]: Software engineers hate this PR!  Find out why...

File: server/interpreter.js [hunk 4]
```diff
@@ -1123,8 +1123,7 @@ Interpreter.prototype.initArray_ = function() {
         if (obj.has(String(k), perms)) {
           var v = obj.get(String(k), perms);
           if (v === searchElement ||
-              (typeof v === 'number' && typeof searchElement === 'number' &&
-               isNaN(v) && isNaN(searchElement))) {
+              (Number.isNaN(v) && Number.isNaN(searchElement))) {
```
Comment (cpcallen) [inline server/interpreter.js]: We should set up a presubmit to make sure the compiler—and tests—have been run:

```
interpreter.js:1118: WARNING - actual parameter 1 of Number.isNaN does not match formal parameter
found   : (boolean|module$interpreter.default.prototype.Object|null|number|string|undefined)
required: number
              (Number.isNaN(v) && Number.isNaN(searchElement))) {
                            ^

interpreter.js:1118: WARNING - actual parameter 1 of Number.isNaN does not match formal parameter
found   : (boolean|module$interpreter.default.prototype.Object|null|number|string|undefined)
required: number
              (Number.isNaN(v) && Number.isNaN(searchElement))) {
                                               ^^^^^^^^^^^^^

0 error(s), 2 warning(s), 78.7% typed
```

**Pull request merged**